packages/react/src/template.tsx

```diff
@@ -13,28 +13,27 @@ export type ComponentDefinition<T extends Record<string, JSONValue>> = {
 }
 
 /** Data state */
-export type DataState<T = never> =
+export type DataState<T = EventDataState> =
```

Comment:
fix: ServerDataState (DataState on the server, before serialisation) now will not have EventDataState. This is correct. EventDataState without function on the server is going to result in an error, so should be properly typed. I will improve this type in a follow-up anyway.